### PR TITLE
feat: [CDS-49860]: fixed service schema checks for TAS

### PR DESCRIPTION
**Pull request by RishabhGupta34**
## Describe your changes
Fixed service schema checks for TAS

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/42375)

### CODE REVIEW

**Review by anilchowdhury-harness — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42375)* status: 5 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[`127-cd-nextgen-entities/src/main/java/io/harness/cdng/tas/TasEntityHelper.java` line 137 at r2](https://reviewable.io/reviews/harness/harness-core/42375#-NLLQxJt7qkTPg_C3d6V:-NLLQxJt7qkTPg_C3d6W:b74c9du) ([raw file](https://github.com/harness/harness-core/blob/f4e167f608b3685d1feaa089b5919f8769722d1c/127-cd-nextgen-entities/src/main/java/io/harness/cdng/tas/TasEntityHelper.java#L137)):*
> ```Java
>       }
>     }
>   }
> ```

lets simplify this method as discussed

**Review by anilchowdhury-harness — DISMISSED** [review]
:lgtm:

> Reviewed 1 of 2 files at r1, 3 of 4 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42375#-:-NLP9u4mArnZtah0xe8p:bnfp4nl)* status: 5 file(s) to review. 1 of 2 approvals obtained.

**Comment by yogesh-chauhan** [comment]
*[`127-cd-nextgen-entities/src/main/java/io/harness/ng/core/service/services/impl/ServiceEntityServiceImpl.java` line 125 at r3](https://reviewable.io/reviews/harness/harness-core/42375#-NLPLNimCTARw7yJmMfo:-NLPLNin3s3rEj91kTbi:b-drse5o) ([raw file](https://github.com/harness/harness-core/blob/cc359841d9ee875eb42e638205e437b276addd3c/127-cd-nextgen-entities/src/main/java/io/harness/ng/core/service/services/impl/ServiceEntityServiceImpl.java#L125)):*
> ```Java
>   private final ServiceOverrideService serviceOverrideService;
>   private final ServiceEntitySetupUsageHelper entitySetupUsageHelper;
>   @Inject private TasEntityHelper tasEntityHelper;
> ```

Lets use factory pattern here. Create a serviceEntityValidatorFactory which gives an instance of TasServicEntityValidator and then call validate(..) method on that
In short, ServiceEntityServiceImpl cannot have any deployment specific logic

**Comment by yogesh-chauhan** [comment]
*[`127-cd-nextgen-entities/src/main/java/io/harness/ng/core/service/services/validators/ServiceEntityValidator.java` line 19 at r4](https://reviewable.io/reviews/harness/harness-core/42375#-NLPetvy7XRBFgssfFoZ:-NLPetvy7XRBFgssfFo_:bw6c60i) ([raw file](https://github.com/harness/harness-core/blob/445a86ec944cbae43cfceac5494dc6bc836e1904/127-cd-nextgen-entities/src/main/java/io/harness/ng/core/service/services/validators/ServiceEntityValidator.java#L19)):*
> ```Java
> @OwnedBy(HarnessTeam.CDP)
> public interface ServiceEntityValidator {
>   boolean supportsValidation();
> ```

do we need this ? the validate method can be no-op for this directly without having to check for this method

**Review by yogesh-chauhan — COMMENTED** [review]
> Reviewed 1 of 2 files at r1, 4 of 8 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42375)* status: 8 file(s) to review. 0 of 2 approvals obtained.

**Review by yogesh-chauhan — DISMISSED** [review]
> Reviewed 1 of 8 files at r4, 4 of 4 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42375)* status: 8 file(s) to review. 1 of 2 approvals obtained.

**Review by anilchowdhury-harness — APPROVED** [review]
:lgtm:

> Reviewed 3 of 8 files at r4, 4 of 4 files at r5, 1 of 1 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42375#-:-NLQ4pvc4Fy1CkhX3KRb:bnfp4nl)* status: 1 file(s) to review. 1 of 2 approvals obtained.

**Review by yogesh-chauhan — APPROVED** [review]
> Reviewed 1 of 1 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42375)* status: :shipit: complete! ✓
